### DISCUSSION TITLE
Error to use jest integration test:    TypeError: Class extends value #<Object> is not a constructor or null

### DISCUSSION THREAD

**Discussion by gustavogmfarias** [prisma/prisma#14504] [Q&A] [answered]
Hello, i'm having problem when i try to test my code with jest.

Anyone passed for the same?

Github: https://github.com/gustavogmfarias/iffolha-js-backend/blob/5118ed1e6ab3cb4559a73949f3ad7e40b2dbadec/prisma/prisma-environment-jest.js#L1-L42

![image](https://user-images.githubusercontent.com/53410368/181096279-3c3a97fd-05cf-4758-b1d2-b2f912d00120.png)

```ts

/* eslint-disable @typescript-eslint/no-var-requires */
const NodeEnvironment = require("jest-environment-node");
const { v4: uuid } = require("uuid");
const { execSync } = require("child_process");
const { resolve } = require("path");
const { Client } = require("pg");

const prismaCli = "./node_modules/.bin/prisma";

require("dotenv").config({
    path: resolve(__dirname, "..", ".env.test"),
});

class CustomEnvironment extends NodeEnvironment {
    constructor(config) {
        super(config);
        this.schema = `code_schema_${uuid()}`;
        console.log("schemas", this.schema);
        this.connectionString = `${process.env.DATABASE_URL}${this.schema}`;
    }

    setup() {
        process.env.DATABASE_URL = this.connectionString;
        this.global.process.env.DATABASE_URL = this.connectionString;

        // Rodar as migrations
        execSync(`${prismaCli} migrate dev`);
    }

    async teardown() {
        const client = new Client({
            connectionString: this.connectionString,
        });

        await client.connect();
        await client.query(`DROP SCHEMA IF EXISTS "${this.schema}" CASCADE`);
        await client.end();
    }
}

module.exports = CustomEnvironment;
```

**Comment by austincrim** [comment, marked as answer]
Hey @gustavogmfarias 👋,

Could you try importing like this:
```ts
const NodeEnvironment = require('jest-environment-node').default
```

I'm looking at some of the code examples [here](https://jestjs.io/docs/configuration#testenvironment-string).

**Reply by gustavogmfarias** [reply]
Hey! Thnks so much, that works, but unfortunately it's still showing errors,.. even thoug it's now about jest only.

**Comment by davidbbe** [comment]
@gustavogmfarias I think this might have to do with the way Jest changed the constructor of the test environment. I can see in your repo that you are using Jest v28. But it looks like you are using the Jest v27 constructor format. Take a look here at what was changed https://jestjs.io/docs/upgrading-to-jest28#test-environment try updating that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
